kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/ConvertUtils.java

```diff
@@ -169,8 +169,11 @@ public static ProcessInstanceCustomVarsList convertToProcessInstanceCustomVarsLi
                 .initiator(pi.getInitiator())
                 .state(pi.getState())
                 .lastModificationDate(pi.getLastModificationDate())
-                .variables(pi.getVariables())
                 .build();
+        
+        if(pi.getVariables() != null && !pi.getVariables().isEmpty()) {
+            instance.setVariables(pi.getVariables());
+        }
 
```

Comment:
I think this is not needed since if it is null or empty and it is set using variables() method, nothing bad happens, does it?